.github/workflows/build.yml

```diff
@@ -0,0 +1,90 @@
+name: build
+on:
+  push:
+    branches:
+    - master
+    - v*
+    tags:
+    - v*
+  pull_request:
+jobs:
+  build:
+    name: Build Binaries
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Codes
+      uses: actions/checkout@v4
+
+    # Build binaries
+    - name: Run CI
+      run: make ci
+
+    # Run e2e test
+    #- name: Run E2e Test
+    #  run: make e2e-test
+
+    # Upload binaries
+    - name: Upload Binaries
+      uses: actions/upload-artifact@v4
+      with:
+        name: binaries_artifact
+        path: ./bin/*
+
+  build_push_image:
+    name: Build and Push Images
+    runs-on: ubuntu-latest
+    needs: build
+    if: ${{ startsWith(github.ref, 'refs/heads/') || startsWith(github.ref, 'refs/tags/') }}
+    steps:
+    - name: Checkout Codes
+      uses: actions/checkout@v4
+
+    - name: Download Binaries
+      uses: actions/download-artifact@v4
+      with:
+        name: binaries_artifact
+        path: ./bin/
+
+    - name: Add Executable Permission
+      run: |
+        chmod +x ./bin/*
+
+    - name: Copy bin Folder to package Folder
+      run: |
+        cp -r ./bin ./package/
+
+    # For multi-platform support
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Declare Branch
+      run: |
+        echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> "$GITHUB_ENV"
+    - name: Login to Docker Hub
```

Comment:
In rancher organization, we may need to read docker credential like this.

https://github.com/harvester/load-balancer-harvester/blob/0025da293f105cc500a27795249e4f952d80adcd/.github/workflows/template-build.yml#L19-L21

https://github.com/harvester/load-balancer-harvester/blob/0025da293f105cc500a27795249e4f952d80adcd/.github/workflows/template-build.yml#L35-L41

Reply:
Modified. Thanks